apis/metal3.io/v1alpha1/baremetalhost_webhook.go

```diff
@@ -32,7 +32,7 @@ func (r *BareMetalHost) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-//+kubebuilder:webhook:verbs=create;update,path=/validate-metal3-io-v1alpha1-baremetalhost,mutating=false,failurePolicy=fail,sideEffects=none,admissionReviewVersions=v1;v1beta1,groups=metal3.io,resources=baremetalhosts,versions=v1alpha1,name=vbaremetalhost.kb.io
+//+kubebuilder:webhook:verbs=create;update,path=/validate-metal3-io-v1alpha1-baremetalhost,mutating=false,failurePolicy=ignore,sideEffects=none,admissionReviewVersions=v1;v1beta1,groups=metal3.io,resources=baremetalhosts,versions=v1alpha1,name=vbaremetalhost.kb.io
```

Comment:
As [discussed previously](https://github.com/metal3-io/baremetal-operator/pull/865#discussion_r622207670), I think that the failure policy should be fail, not ignore.

Reply:
This is the reason of WIP label. Because there seem to be a couple of bugs in different places when we set FailurePolicy to fail. This PR https://github.com/metal3-io/metal3-dev-env/pull/683 is prerequisite to integration tests pass and switch back to FailurePolicy to fail.